src/main/java/ua/antibyte/cinema/service/security/AuthenticationServiceImpl.java

```diff
@@ -22,13 +21,10 @@ public AuthenticationServiceImpl(UserService userService,
     @Override
     public User login(String email, String password)
             throws AuthenticationException {
-        Optional<User> optionalUser = userService.findByEmail(email);
-        if (optionalUser.isPresent()) {
-            User user = optionalUser.get();
-            String hashPassword = HashUtil.hashPassword(password, user.getSalt());
-            if (user.getPassword().equals(hashPassword)) {
-                return user;
-            }
+        User user = userService.findByEmail(email);
```

Comment:
the problem is that previously, if user entered incorrect email, the AuthenticationException was thrown, and now - NoSuchElementException

I would suggest rather to call optional.get() in UserController
what do you think?